### PR TITLE
Add FindValidNodes() to statdb

### PR DESCRIPTION
**Pull request by mobyvb**
https://storjlabs.atlassian.net/browse/V3-398

### CODE REVIEW

File: pkg/storage/segments/store.go [hunk 1]
```diff
@@ -113,6 +113,7 @@ func (s *segmentStore) Put(ctx context.Context, data io.Reader, expiration time.
 		}
 	} else {
 		// uses overlay client to request a list of nodes
+		// TODO(moby) update Space
```
Comment (navillasa) [inline pkg/storage/segments/store.go]: what is update Space?

Comment (mobyvb) [reply pkg/storage/segments/store.go]: It's the param passed into the overlay below. I'll update the comment to be clearer

File: pkg/statdb/statdb_test.go [hunk 2]
```diff
@@ -120,6 +120,67 @@ func TestGetDoesNotExist(t *testing.T) {
...
+		totalUptimeCount   int64
+		uptimeRatio        float64
+	}{
+		{[]byte("id1"), 10, 20, 0.5, 10, 20, 0.5},   // bad ratios
+		{[]byte("id2"), 20, 20, 1, 20, 20, 1},       // good ratios
+		{[]byte("id3"), 20, 20, 1, 10, 20, 0.5},     // good audit success bad uptime
+		{[]byte("id4"), 10, 20, 0.5, 20, 20, 1},     // good uptime bad audit success
+		{[]byte("id5"), 5, 5, 1, 5, 5, 1},           // good ratios not enough audits
+		{[]byte("id6"), 20, 20, 1, 20, 20, 1},       // good ratios, excluded from query
+		{[]byte("id7"), 19, 20, 0.95, 19, 20, 0.95}, // borderline ratios
```
[the remark this answers is not in the document]
Comment (mobyvb) [reply pkg/statdb/statdb_test.go]: It means it's in statdb, but it won't be passed into the id field of FindValidNodes

File: pkg/statdb/statdb.go [hunk 3]
```diff
@@ -119,6 +120,74 @@ func (s *Server) Get(ctx context.Context, getReq *pb.GetRequest) (resp *pb.GetRe
...
+func getQueryString(nodeIds [][]byte, auditCount int64, auditSuccess, uptime float64) string {
+	idStr := "("
+	for i, id := range nodeIds {
+		idStr += fmt.Sprintf(`"%s"`, string(id))
+		if i+1 < len(nodeIds) {
+			idStr += ","
+		}
+	}
+	idStr += ")"
+	queryStr := fmt.Sprintf(`SELECT nodes.id, nodes.total_audit_count, nodes.audit_success_ratio, nodes.uptime_ratio, nodes.created_at
```
[the remark this answers is not in the document]
Comment (navillasa) [reply pkg/statdb/statdb.go]: what would be the right way to do this? using this package? https://godoc.org/database/sql

[the remark this answers is not in the document]
Comment (egonelbre) [reply pkg/statdb/statdb.go]: `db.QueryRow("SELECT * FROM Users WHERE id in ?", nodeids)`, assuming the binding variable is `?` and it supports arrays. 
There are examples in https://godoc.org/database/sql#example-Rows. And a full tutorial in http://go-database-sql.org/retrieving.html

**Pull request merged**